### ISSUE TITLE
Decision splitting for unsigned char

### ISSUE THREAD

**Issue by rajdeep87** [diffblue/2ls#25]
Consider the structure in regression "FIFO":
struct state_elements_rbFIFO{                                                 
  _Bool empty;                                                                
  unsigned char mem[16];                                                      
  unsigned char head;                                                         
  unsigned char tail;                                                         
};                                                                            
struct state_elements_rbFIFO  srbFIFO;                                        
    
ACDL cannot split decision variable srbFIFO.mem because its type is unsigned char (warning below).
WARNING: do not know how to split srbFIFO.mem#166 of type unsigned char [16l]

Need to support splitting for unsigned char.

**Comment by peterschrammel** [comment]
The problem is not the unsigned char. It is the array (type unsigned char [16l])

**Comment by rajdeep87** [comment]
Exactly. Each time an array cell is accessed for write operation, the SSA generates a new instance of the variable. For example, the following different instances are created for the array variable "mem"  
srbFIFO.mem#166  ,                                                                        
srbFIFO.mem#22  ,                                                                         
srbFIFO.mem#39  ,
ssrFIFO.mem#105  ,
ssrFIFO.mem#106  ,
ssrFIFO.mem#23  ,                                                                         
ssrFIFO.mem#48
